src/main/java/teammates/ui/pagedata/InstructorStudentRecordsAjaxPageData.java

```diff
@@ -13,18 +14,24 @@
     private List<FeedbackResultsTable> resultsTables;
 
     public InstructorStudentRecordsAjaxPageData(AccountAttributes account, StudentAttributes student, String sessionToken,
-                                                List<FeedbackSessionResultsBundle> results) {
+                                                List<FeedbackSessionResultsBundle> results,
+                                                Map<String, Boolean> sessionSubmissionStatusMap) {
         super(account, student, sessionToken);
         this.resultsTables = new ArrayList<>();
         for (int i = 0; i < results.size(); i++) {
             FeedbackSessionResultsBundle result = results.get(i);
+            boolean submissionStatus = true;
+            try {
+                submissionStatus = sessionSubmissionStatusMap.get(result.getFeedbackSession().getFeedbackSessionName());
+            } catch (Exception e) {
```

Comment:
Why do we need this? This looks dangerous.

Reply:
This will never go to catch block but I just kept it for going by the standard way of accessing map and handling cases when key is not found.

Reply:
Why would you check a key-value pair absence by catching an Exception? That is a rather dangerous approach.